### ISSUE TITLE
🚀 Add a SPA mode to FileSystem middleware

### ISSUE THREAD

**Issue by AzSiAz** [gofiber/fiber#708]
**Is your feature request related to a problem?**
New middleware Filesystem can't serve an spa
 
**Describe the solution you'd like**
Something like this automatically redirecting to index.html on file not found
```
app.Use("/", middleware.FileSystem(middleware.FileSystemConfig{
  Root:   pkger.Dir("/web/dist"), // http.Dir("/web/dist")
  Index:  "index.html",
  Browse: false,
  Spa: true, // or SendToIndexOnFileNotFound to be more explicit
}))
```

**Describe alternatives you've considered**
Good old `Get("/*")` to send index.html on not found but I'm mostly duplicating FileSystem middleware

**Additional context**
Embed/Serve a React/Angular/Vue/other
Code is here: https://github.com/AzSiAz/Odin

**Comment by Fenny** [comment]
You could try the following
```go
mw := middleware.FileSystem(middleware.FileSystemConfig{
  Root:    pkger.Dir("/web/dist"),
  Index:  "index.html",
  Browse: false,
})
app.Use("/", mw)
app.Use("*", mw)
```

**Comment by AzSiAz** [comment]
did not think of that one, but it's doesn't seem to fallback to index.html either

**Comment by Fenny** [comment]
I updated my example, could you try that?

**Comment by AzSiAz** [comment]
Same error message `file does not exist`
By the way I'm on version 1.13.3

**Comment by AzSiAz** [comment]
First post edited with code link

**Comment by ReneWerner87** [comment]
test with https://github.com/SinghDigamber/angular-routing-tutorial.git

```go
git clone https://github.com/SinghDigamber/angular-routing-tutorial.git
cd angular-routing-tutorial
npm i
npm run build
go mod init angular-routing-tutorial

```
create main.go and run the spa
```go
package main

import (
  "github.com/gofiber/fiber"
  "log"
)

func main() {
  app := fiber.New()
  app.Static("/", "./dist/angular-routing-tutorial")
  app.Get("/*", func(ctx *fiber.Ctx) {
    ctx.SendFile("./dist/angular-routing-tutorial/index.html")
  })
  log.Fatal(app.Listen(4200))
}
```

as described in the examples
https://github.com/gofiber/recipes/blob/master/spa/main.go

@AzSiAz since only 2 simple lines are needed here, I would not create extra middleware

![image](https://user-images.githubusercontent.com/7063188/91706458-b5662b80-eb7e-11ea-8b97-8c403af1f584.png)
![image](https://user-images.githubusercontent.com/7063188/91706471-bc8d3980-eb7e-11ea-9271-36ddc860fb0c.png)
![image](https://user-images.githubusercontent.com/7063188/91706496-c44cde00-eb7e-11ea-96d8-df659e63d6fa.png)

**Comment by AzSiAz** [comment]
Sorry but that's not a one to one implementation since I am using FileSystem middleware with assets embedding in my exemple

**Comment by Fenny** [comment]
@AzSiAz, the feature will be available in [v1.15.0](https://gofiber.io/v15) 👍

**Comment by AzSiAz** [comment]
Thanks 😁
